clicache/integration-test/AttributesFactoryTestsN.cs

```diff
@@ -15,94 +15,29 @@
-//      RegionAttributes<object, object> attrs;
-//      AttributesFactory<object, object> af = new AttributesFactory<object, object>();
-//      af.SetScope(ScopeType.Local);
-//      af.SetEndpoints("bass:1234");
-//      attrs = af.CreateRegionAttributes();
-//      region = CacheHelper.CreateRegion<object, object>("region1", attrs);
-//      try
-//      {
-//       IRegion<Object, Object> localRegion = region.GetLocalView();
-//        Assert.Fail(
-//          "LOCAL scope is incompatible with a native client region");
-//      }
-//      catch (UnsupportedOperationException)
-//      {
-//        Util.Log("Got expected UnsupportedOperationException for " +
-//          "LOCAL scope for native client region");
-//      }
-
-//      af.SetScope(ScopeType.Local);
-//      af.SetClientNotificationEnabled(true);
-//      attrs = af.CreateRegionAttributes();
-//      try
-//      {
-//        region = CacheHelper.CreateRegion<object, object>("region2", attrs);
-//        Assert.Fail(
-//          "LOCAL scope is incompatible with clientNotificationEnabled");
-//      }
-//      catch (UnsupportedOperationException)
-//      {
-//        Util.Log("Got expected UnsupportedOperationException for " +
-//          "clientNotificationEnabled for non native client region");
-//      }
-
-//      // Checks for HA regions
-
-//      CacheHelper.CloseCache();
-//      af.SetScope(ScopeType.Local);
-//      af.SetEndpoints("bass:3434");
-//      af.SetClientNotificationEnabled(false);
-//      attrs = af.CreateRegionAttributes();
-//      try
-//      {
-//        region = CacheHelper.CreateRegion<object, object>("region2", attrs);
-//        Assert.Fail(
-//          "LOCAL scope is incompatible with a native client HA region");
-//      }
-//      catch (UnsupportedOperationException)
-//      {
-//        Util.Log("Got expected UnsupportedOperationException for " +
-//          "LOCAL scope for native client region");
-//      }
-
-//      af.SetScope(ScopeType.Local);
-//      af.SetEndpoints("none");
-//      af.SetClientNotificationEnabled(false);
-//      attrs = af.CreateRegionAttributes();
-//      region = CacheHelper.CreateRegion<object, object>("region1", attrs);
-//      Util.Log("C++ local region created with HA cache specification.");
-//    }
-//  }
-//}
+using System;
+using System.Reflection;
+
+#pragma warning disable 618
+
+namespace Apache.Geode.Client.UnitTests
+{
+  using NUnit.Framework;
+  using Apache.Geode.DUnitFramework;
+  using Apache.Geode.Client;
+
+  [TestFixture]
+  [Category("group1")]
+  [Category("unicast_only")]
+  [Category("generics")]
+  public class AttributesFactoryTests : UnitTests
+  {
+    [Test]
+    public void fluentModeltest()
+    {
+      AttributesFactory<TKey, TValue> af = new AttributesFactory<TKey, TValue>();
+      Apache.Geode.Client.RegionAttributes<TKey, TValue> rattrs = af.SetLruEntriesLimit(0).SetInitialCapacity(5).CreateRegionAttributes();
+      Assert.IsNotNull(rattrs);
```

Comment:
I would suggest adding a bit more to this test. If the point is to assert that the return from each set call is the same object you called the setter an then I would put that in the test. All this test does is show that the last call creates a region attribute. The intermediate set calls could be returning different factory objects.

Reply:
Good point.